web/src/lib/components/shared-components/combobox.svelte

```diff
@@ -11,81 +11,198 @@
+    if (filteredOptions.length === 0) {
+      selectedIndex = 0;
+    } else if (selectedIndex === undefined) {
+      selectedIndex = increment === 1 ? 0 : filteredOptions.length - 1;
+    } else {
+      selectedIndex = (selectedIndex + increment + filteredOptions.length) % filteredOptions.length;
     }
+    await tick();
+    optionRefs[selectedIndex]?.scrollIntoView({ block: 'nearest' });
+  };
+
+  const onInput: FormEventHandler<HTMLInputElement> = (event) => {
+    openDropdown();
+    searchQuery = (event.target as HTMLInputElement).value;
+    selectedIndex = undefined;
+    optionRefs[0]?.scrollIntoView({ block: 'nearest' });
   };
 
-  let handleSelect = (option: ComboBoxOption) => {
+  let onSelect = (option: ComboBoxOption) => {
     selectedOption = option;
+    searchQuery = option.label;
     dispatch('select', option);
-    isOpen = false;
+    closeDropdown();
   };
 
   const onClear = () => {
     selectedOption = undefined;
     searchQuery = '';
     dispatch('select', selectedOption);
   };
+
+  const onClick = (event: MouseEvent) => {
+    const target = event.target as HTMLElement;
+    if (comboboxRef && !comboboxRef.contains(target)) {
+      deactivate();
+    }
+  };
 </script>
 
-<div class="relative w-full dark:text-gray-300 text-gray-700 text-base" use:clickOutside on:outclick={handleOutClick}>
+<label class="text-sm text-black dark:text-white" class:sr-only={hideLabel} for={inputId}>{label}</label>
+<div class="relative w-full dark:text-gray-300 text-gray-700 text-base" bind:this={comboboxRef}>
   <div>
-    {#if isOpen}
+    {#if isActive}
       <div class="absolute inset-y-0 left-0 flex items-center pl-3">
         <div class="dark:text-immich-dark-fg/75">
-          <Icon path={mdiMagnify} />
+          <Icon path={mdiMagnify} ariaHidden={true} />
         </div>
       </div>
     {/if}
 
     <input
-      {id}
       {placeholder}
-      role="combobox"
+      aria-activedescendant={selectedIndex || selectedIndex === 0 ? `${listboxId}-${selectedIndex}` : ''}
+      aria-autocomplete="list"
+      aria-controls={listboxId}
       aria-expanded={isOpen}
-      aria-controls={id}
-      class="immich-form-input text-sm text-left w-full !pr-12 transition-all"
-      class:!pl-8={isOpen}
+      autocomplete="off"
+      class:!pl-8={isActive}
       class:!rounded-b-none={isOpen}
-      class:cursor-pointer={!isOpen}
-      value={isOpen ? '' : selectedOption?.label || ''}
-      on:input={(e) => (searchQuery = e.currentTarget.value)}
-      on:focus={handleClick}
-      on:blur={() => (inputFocused = false)}
+      class:cursor-pointer={!isActive}
+      class="immich-form-input text-sm text-left w-full !pr-12 transition-all"
+      id={inputId}
+      on:click={activate}
+      on:focus={activate}
+      on:input={onInput}
+      role="combobox"
+      type="text"
+      value={searchQuery}
+      use:shortcuts={[
+        {
+          shortcut: { key: 'Tab' },
```

Comment:
Don't know how reliable this detection is, because macos safari has an alternative shortcut for moving focus if I recall correctly. Maybe we could use something like this on the outer div and also prevent duplicating outclick logic:

```svelte
<div
  use:clickOutside={{ onOutclick: deactivate }}
  on:focusout={(e) => {
    if (e.relatedTarget instanceof Node && !e.currentTarget.contains(e.relatedTarget)) {
      deactivate();
    }
  }}
```

Reply:
This approach is even simpler, nice!